src/main/java/ua/tc/marketplace/service/CategoryService.java

```diff
@@ -2,42 +2,102 @@
 
 
 import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import ua.tc.marketplace.exception.model.CategoryCreationException;
+import ua.tc.marketplace.exception.model.CategoryNotFoundException;
+import ua.tc.marketplace.exception.model.MappingException;
 import ua.tc.marketplace.model.entity.Category;
 import ua.tc.marketplace.repository.CategoryRepository;
+import ua.tc.marketplace.model.dto.CategoryDto;
+import ua.tc.marketplace.util.mapper.CategoryMapper;
+import ua.tc.marketplace.util.mapper.ClassificationAttributeMapper;
 
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class CategoryService {
-
+    @Autowired
```

Comment:
remove `@Autowired` . Spring automatically injects beans when we use parametrized Constructor. You use `@RequiredArgsConstructor` and it is enough